interfaces/apparmor/template.go

```diff
@@ -952,11 +952,11 @@ profile snap-update-ns.###SNAP_INSTANCE_NAME### (attach_disconnected) {
   # Allow reading the dynamic linker cache.
   /etc/ld.so.cache r,
   # Allow reading, mapping and executing the dynamic linker.
-  /{,usr/}lib{,32,64,x32}/{,@{multiarch}/{,atomics/}}ld-*.so mrix,
+  /{,{,var/lib/snapd/}snap/{snapd,core}/*/}{,usr/}lib{,32,64,x32}/{,@{multiarch}/{,atomics/}}ld-*.so* mrix,
   # Allow reading and mapping various parts of the standard library and
   # dynamically loaded nss modules and what not.
-  /{,usr/}lib{,32,64,x32}/{,@{multiarch}/{,atomics/}}libc{,-[0-9]*}.so* mr,
-  /{,usr/}lib{,32,64,x32}/{,@{multiarch}/{,atomics/}}libpthread{,-[0-9]*}.so* mr,
+  /{,{,var/lib/snapd/}snap/{snapd,core}/*/}{,usr/}lib{,32,64,x32}/{,@{multiarch}/{,atomics/}}libc{,-[0-9]*}.so* mr,
+  /{,{,var/lib/snapd/}snap/{snapd,core}/*/}{,usr/}lib{,32,64,x32}/{,@{multiarch}/{,atomics/}}libpthread{,-[0-9]*}.so* mr,
```

Comment:
do we need these for snapctl and similar?

Reply:
snapctl, snap-exec and snap-update-ns should be static... :thinking: 

I will try to figure out what failed without it.

Reply:
After some testing, it seems this is not needed. It might have been that I have tried to make these binaries not static. But back-tracked. I will try to do that in a different PR. And I will revert the changes of this file once the I have run enough tests in #13517

Reply:
I did not see any failure related to that. So I reverted it.